resilience4j-spring/src/main/java/io/github/resilience4j/circuitbreaker/configure/CircuitBreakerAspect.java

```diff
@@ -38,89 +42,126 @@
+		ApiType type = backendMonitored.type();
+		io.github.resilience4j.circuitbreaker.CircuitBreaker circuitBreaker = getOrCreateCircuitBreaker(methodName, backend);
+		return handleJoinPoint(proceedingJoinPoint, circuitBreaker, methodName, type);
+	}
+
+	private io.github.resilience4j.circuitbreaker.CircuitBreaker getOrCreateCircuitBreaker(String methodName, String backend) {
+		io.github.resilience4j.circuitbreaker.CircuitBreaker circuitBreaker = circuitBreakerRegistry.circuitBreaker(backend,
+				() -> circuitBreakerProperties.createCircuitBreakerConfig(backend));
+
+		if (logger.isDebugEnabled()) {
+			logger.debug("Created or retrieved circuit breaker '{}' with failure rate '{}' and wait interval'{}' for method: '{}'",
+					backend, circuitBreaker.getCircuitBreakerConfig().getFailureRateThreshold(),
+					circuitBreaker.getCircuitBreakerConfig().getWaitDurationInOpenState(), methodName);
+		}
+
+		return circuitBreaker;
+	}
+
+	private CircuitBreaker getBackendMonitoredAnnotation(ProceedingJoinPoint proceedingJoinPoint) {
+		if (logger.isDebugEnabled()) {
+			logger.debug("circuitBreaker parameter is null");
+		}
+		CircuitBreaker circuitBreaker = null;
+		Class<?> targetClass = proceedingJoinPoint.getTarget().getClass();
+		if (targetClass.isAnnotationPresent(CircuitBreaker.class)) {
+			circuitBreaker = targetClass.getAnnotation(CircuitBreaker.class);
+			if (circuitBreaker == null && logger.isDebugEnabled()) {
+				logger.debug("TargetClass has no annotation 'CircuitBreaker'");
+				circuitBreaker = targetClass.getDeclaredAnnotation(CircuitBreaker.class);
+				if (circuitBreaker == null && logger.isDebugEnabled()) {
+					logger.debug("TargetClass has no declared annotation 'CircuitBreaker'");
+				}
+			}
+		}
+		return circuitBreaker;
+	}
+
+	private Object handleJoinPoint(ProceedingJoinPoint proceedingJoinPoint, io.github.resilience4j.circuitbreaker.CircuitBreaker circuitBreaker, String methodName, ApiType type) throws Throwable {
+		if (type == ApiType.WEBFLUX) {
+			return defaultWebFlux(proceedingJoinPoint, circuitBreaker, methodName);
+		} else {
+			return defaultHandling(proceedingJoinPoint, circuitBreaker, methodName);
+		}
+	}
+
+	/**
+	 * handle the Spring web flux (Flux /Mono) return types AOP based into reactor circuit-breaker
+	 * See {@link io.github.resilience4j.reactor.circuitbreaker.operator.CircuitBreakerOperator} for details.
+	 */
+	private Object defaultWebFlux(ProceedingJoinPoint proceedingJoinPoint, io.github.resilience4j.circuitbreaker.CircuitBreaker circuitBreaker, String methodName) throws Throwable {
+		CircuitBreakerUtils.isCallPermitted(circuitBreaker);
+		long start = System.nanoTime();
+		try {
+			Object returnValue = proceedingJoinPoint.proceed();
+			if (returnValue instanceof Flux) {
+				Flux fluxReturnValue = (Flux) returnValue;
+				return fluxReturnValue.transform(CircuitBreakerOperator.of(circuitBreaker));
+			} else if (returnValue instanceof Mono) {
+				Mono fluxReturnValue = (Mono) returnValue;
```

Comment:
variable name here says flux but is a mono

Reply:
updated !